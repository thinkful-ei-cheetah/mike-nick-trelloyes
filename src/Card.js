import React from 'react';
import './Card.css';

class Card extends React.Component {
  render() {
    return (
      <div className='Card'>
        <h3>{this.props.title}</h3>
        <p>{this.props.content}</p>
      </div>
    )
  }
}

export default Card;