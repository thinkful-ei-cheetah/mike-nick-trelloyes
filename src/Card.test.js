import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card Component', () => {
  it('it should render successfully', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card/>, div)
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('when props given', () => {
    it('it displays title and content', () => {
      const tree = renderer.create(
        <Card title='foo' content='bar' />
      ).toJSON()
      expect(tree).toMatchSnapshot();
    });
  })
  
  describe('when NO props given', () => {
    it('should render empty if no title or content is given', () => {
      const tree = renderer.create(
        <Card />
      ).toJSON()
      expect(tree).toMatchSnapshot();
    })
  })
})