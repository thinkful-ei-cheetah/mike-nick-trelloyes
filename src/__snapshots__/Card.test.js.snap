// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component when NO props given should render empty if no title or content is given 1`] = `
<div
  className="Card"
>
  <h3 />
  <p />
</div>
`;

exports[`Card Component when props given it displays title and content 1`] = `
<div
  className="Card"
>
  <h3>
    foo
  </h3>
  <p>
    bar
  </p>
</div>
`;
